### ISSUE TITLE
ApiCompat no longer passes IntermediateOutputPath

### ISSUE THREAD

**Issue by sbomer** [dotnet/arcade#5361]
- [x] This issue is blocking arcade update in mono/linker
- [ ] This issue is causing unreasonable pain

An arcade [update](https://github.com/mono/linker/pull/1144) in mono/linker is failing on ApiCompat.

It looks like ApiCompat was changed to no longer pass `IntermediateOutputPath`: https://github.com/dotnet/arcade/commit/d6dd7a000fe9908e4caae5eb9aa30ecc9cb917ff#diff-3c3f2c92eeec9b69dac2deccfac46603L72, so it isn't resolving the implementation assembly.

@dougbu is this expected? If so, what's the supported way to tell ApiCompat where to find the `IntermediateAssembly`?

**Comment by dougbu** [comment]
Yes, this is expected. We were not aware of scenarios in which `$(IntermediateOutputPath)` mattered. What is the case @ericstj and I missed?

/fyi the previous conversation starts with https://github.com/dotnet/arcade/pull/4585#discussion_r399592856

**Comment by sbomer** [comment]
The scenario is an implementation `csproj` that has a ProjectReference to its `ContractProject` (with `ReferenceOutputAssembly` `false`, and `OutputItemType` `ResolvedMatchingContract`) for comparison. In this case `ReferencePath` doesn't contain the intermediate assembly.

I'm happy to change how we invoke it - that was just how I saw others (dotnet/runtime IIRC) doing it at the time.

**Comment by dougbu** [comment]
@ericstj how do you prefer we handle this?

**Comment by ericstj** [comment]
@dougbu my suggestions was based on `--contract-depends`, not `--impl-dirs`.  In the case of `--contract-depends` `IntermediateOutputPath` is not needed since the actual contract is passed in.  In the case of `--impl-dirs` it is required.  I think you should undo your change to `--impl-dirs`

**Comment by dougbu** [comment]
@ericstj I'm fine reverting that change, restoring `$(IntermediateOutputPath)` in the forward comparison's `--impl-dirs`. However, that option is _exactly_ the same as `--contract-depends` in the reversed comparison. I'll fix things up in both (tomorrow).

**Comment by dougbu** [comment]
Opened #5373.

@sbomer do you have a way to test this out locally before we get this in and so on? For example, changing files in your packages folder with the two-word update should work.

**Comment by sbomer** [comment]
@dougbu thanks! Yes, I've confirmed that the fix works locally. I also confirmed that just the `ApiCompatArgs` part of the fix is enough for our scenario.

**Comment by ViktorHofer** [comment]
Closing as APICompat got removed from arcade, rewritten and now lives in dotnet/sdk: https://github.com/dotnet/sdk/tree/main/src/Compatibility
